oauth2-server/src/main/java/com/clouway/oauth2/CodeExchangeVerificationFlow.java

```diff
@@ -0,0 +1,43 @@
+package com.clouway.oauth2;
+
+import com.clouway.friendlyserve.Request;
+import com.clouway.friendlyserve.Response;
+import com.clouway.oauth2.authorization.Authorization;
+import com.clouway.oauth2.authorization.ClientAuthorizationRepository;
+import com.clouway.oauth2.client.Client;
+import com.clouway.oauth2.codechallenge.CodeVerifier;
+import com.google.common.base.Optional;
+
+/**
+ * @author Vasil Mitov <vasil.mitov@clouway.com>
+ */
+public class CodeExchangeVerificationFlow implements ClientActivity {
```

Comment:
As code exchange requires Authorization for verification it may implement: VerifiedActivity

About the name: VerifiedActivity could become: AuthorizedClientActivity 

```java
Response execute(Authorization authorization, Client client, Request request, DateTime instant);
```

Reply:
I was thinking about that. It becomes a little over the top with naming. We already have AuthorizedClientActivity with construction: 
```java
interface AuthorizedClientActivity {
Response execute(Client client, Identity identity, Set<String> scopes, Request request, DateTime instant, Map<String, String> params);
}
```

Which is responsible for issuing the token.

Reply:
This could become AuthorizedIdentityActivity as identity is passed.

Reply:
Another possible way could be IssueTokenActivity/BearerTokenActivity as the result is BearerTokenResponse

Reply:
You always have to return commonly named Response object and it's not explicitly said that BearertokenResponse is returned.

Reply:
I agree i kept the AuthorizedIdentityActivity